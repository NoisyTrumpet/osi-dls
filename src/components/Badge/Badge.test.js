import React from 'react';
import { render, screen } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveClass,
  toHaveStyle,
  toHaveAttribute,
} from '@testing-library/jest-dom/matchers';
import { Badge } from './';

expect.extend({
  toBeInTheDocument,
  toHaveClass,
  toHaveStyle,
  toHaveAttribute,
});

describe('<Badge />', () => {
  // Renders
  test('renders', () => {
    const { container } = render(<Badge count={1} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  // Correct Class
  test('has correct class and color', () => {
    const { rerender } = render(<Badge count={3} />);
    expect(screen.getByText('3').closest('div')).toHaveClass('badge');
    rerender(<Badge count={4} color="accentA" />);
    expect(screen.getByText('4').closest('div')).toHaveStyle({
      backgroundColor: 'rgb(255, 221, 117);',
    });
  });
  // Does not render if less than 1
  test('does not render if less than 1', () => {
    const { container } = render(<Badge count={0} />);
    expect(container.firstChild).not.toBeInTheDocument();
  });
});
