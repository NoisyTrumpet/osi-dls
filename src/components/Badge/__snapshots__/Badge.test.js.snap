// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders 1`] = `
<div
  class="badge background-color-primary"
  color="primary"
>
  <span
    class="typography typography-caption color-brand-white"
  >
    1
  </span>
</div>
`;
